sdkproject/Assets/Mapbox/Unity/MeshGeneration/LayerVisualizers/VectorLayerVisualizer.cs

```diff
@@ -215,91 +322,123 @@ private IEnumerator ProcessLayer(VectorTileLayer layer, UnityTile tile, Action c
 				yield break;
 			}
 
-			//testing each feature with filters
-			var fc = layer.FeatureCount();
+			Debug.Log("Tile Id -> " + tile.name + "Layer Name -> " + layer.Name + " features -> " + layer.FeatureCount());
+			if (!_vectorFeaturesPerTile.ContainsKey(tile))
+			{
+				_vectorFeaturesPerTile.Add(tile, new VectorLayerVisualizerProperties
+				{
+					vectorTileLayer = layer,
+				});
+			}
+
 			//Get all filters in the array.
-			var filters = _layerProperties.filterOptions.filters.Select(m => m.GetFilterComparer()).ToArray();
+			_vectorFeaturesPerTile[tile].layerFeatureFilters = _layerProperties.filterOptions.filters.Select(m => m.GetFilterComparer()).ToArray();
 
 			// Pass them to the combiner
-			Filters.ILayerFeatureFilterComparer combiner = new Filters.LayerFilterComparer();
+			_vectorFeaturesPerTile[tile].layerFeatureFilterCombiner = new Filters.LayerFilterComparer();
 			switch (_layerProperties.filterOptions.combinerType)
 			{
 				case Filters.LayerFilterCombinerOperationType.Any:
-					combiner = Filters.LayerFilterComparer.AnyOf(filters);
+					_vectorFeaturesPerTile[tile].layerFeatureFilterCombiner = Filters.LayerFilterComparer.AnyOf(_vectorFeaturesPerTile[tile].layerFeatureFilters);
 					break;
 				case Filters.LayerFilterCombinerOperationType.All:
-					combiner = Filters.LayerFilterComparer.AllOf(filters);
+					_vectorFeaturesPerTile[tile].layerFeatureFilterCombiner = Filters.LayerFilterComparer.AllOf(_vectorFeaturesPerTile[tile].layerFeatureFilters);
 					break;
 				case Filters.LayerFilterCombinerOperationType.None:
-					combiner = Filters.LayerFilterComparer.NoneOf(filters);
+					_vectorFeaturesPerTile[tile].layerFeatureFilterCombiner = Filters.LayerFilterComparer.NoneOf(_vectorFeaturesPerTile[tile].layerFeatureFilters);
 					break;
 				default:
 					break;
 			}
 
-			for (int i = 0; i < fc; i++)
-			{
-
-				var buildingsWithUniqueIds =
-					(_layerProperties.honorBuildingIdSetting) && _layerProperties.buildingsWithUniqueIds;
+			_vectorFeaturesPerTile[tile].buildingsWithUniqueIds = (_layerProperties.honorBuildingIdSetting) && _layerProperties.buildingsWithUniqueIds;
 
-				var feature = new VectorFeatureUnity(layer.GetFeature(i), tile, layer.Extent, buildingsWithUniqueIds);
-
-				//skip existing features, only works on tilesets with unique ids
-				if (buildingsWithUniqueIds && _activeIds.Contains(feature.Data.Id))
-				{
-					continue;
-				}
-				else
+			////find any replacement criteria and assign them
+			foreach (var goModifier in _defaultStack.GoModifiers)
+			{
+				if (goModifier is IReplacementCriteria &&
+				  goModifier.Active)
```

Comment:
This could go on line 359 rather than being split into another line

Reply:
Done!